lib/iris/tests/unit/analysis/stats/test_pearsonr.py

```diff
@@ -0,0 +1,126 @@
+# (C) British Crown Copyright 2014 - 2015, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for the `iris.analysis.stats.pearsonr` function."""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+import numpy as np
+import numpy.ma as ma
+
+import iris
+import iris.analysis.stats as stats
+from iris.exceptions import CoordinateNotFoundError
+
+
+@tests.skip_data
+class Test(tests.IrisTest):
+    def setUp(self):
+        # 3D cubes:
+        cube_temp = iris.load_cube(tests.get_data_path(
+            ('NetCDF', 'global', 'xyt', 'SMALL_total_column_co2.nc')))
+        self.cube_a = cube_temp[0:6]
+        self.cube_b = cube_temp[20:26]
+        cube_temp = self.cube_a.copy()
+        cube_temp.coord('latitude').guess_bounds()
+        cube_temp.coord('longitude').guess_bounds()
+        self.weights = iris.analysis.cartography.area_weights(cube_temp)
+
+    def test_perfect_corr(self):
+        r = stats.pearsonr(self.cube_a, self.cube_a,
+                           ['latitude',   'longitude'])
```

Comment:
And while your here you could remove the extra space after this comma so there is just 1 I think...